src/compiler/checker.ts

```diff
@@ -9072,8 +9072,9 @@ namespace ts {
                     // If the parent is a tuple type, the rest element has a tuple type of the
                     // remaining tuple element types. Otherwise, the rest element has an array type with same
                     // element type as the parent type.
-                    type = everyType(parentType, isTupleType) ?
-                        mapType(parentType, t => sliceTupleType(t as TupleTypeReference, index)) :
+                    const resolvedParentType = getBaseConstraintOrType(parentType);

```

Comment:
nit: I would call this `baseConstraint` or smth. `resolvedType` is used in the codebase almost only in the context of `NodeLinks` - so `resolvedParentType` could be confused with that.

Reply:
Good point, thanks!